### ISSUE TITLE
Write Permission denied for ~/.cache/SpaceVim/conf/init.json

### ISSUE THREAD

**Issue by bharathyes** [wsdjeg/SpaceVim#4715]
## Expected behavior:

`~/.cache/SpaceVim/conf/init.json` file is automatically created with appropriate permissions.

## The reproduce ways from Vim starting (Required!):

Use nvim to open any file. Error displayed every time `spacevim` is launched.

Made a new install of both nvim and spacevim:
- extract nvim tar, move to /opt and add to path.
- install spacevim using provided script
- launch nvim

PS: also was showing permission issue for `~/.cache/SpaceVim` which got resolved by #4714

## Output of the `:SPDebugInfo!`

<details><summary> SpaceVim debug information </summary>

### SpaceVim options :

```toml
  auto_disable_touchpad = 1
  autocomplete_method = 'deoplete'
  autocomplete_parens = 1
  buffer_index_type = 4
  checkinstall = 1
  code_runner_focus = 0
  colorscheme = 'gruvbox'
  colorscheme_bg = 'dark'
  colorscheme_default = 'desert'
  commandline_prompt = '>'
  custom_color_palette = []
  custom_plugins = []
  data_dir = '/home/shogun/.cache/'
  debug_level = 1
  default_custom_leader = '<Space>'
  default_indent = 2
  disabled_plugins = []
  enable_bepo_layout = 0
  enable_cursorcolumn = 0
  enable_cursorline = 1
  enable_debug = 0
  enable_googlesuggest = 0
  enable_guicolors = 0
  enable_javacomplete2_py = 0
  enable_key_frequency = 0
  enable_language_specific_leader = 1
  enable_neocomplcache = 0
  enable_os_fileformat_icon = 0
  enable_powerline_fonts = 1
  enable_projects_cache = 1
  enable_statusline_bfpath = 0
  enable_statusline_mode = 0
  enable_statusline_tag = 1
  enable_tabline_ft_icon = 0
  enable_vimfiler_welcome = 1
  enable_ycm = 0
  error_symbol = '✖'
  escape_key_binding = 'jk'
  expand_tab = 1
  file_searching_tools = []
  filemanager = 'nerdtree'
  filetree_direction = 'right'
  filetype_icons = {}
  force_global_config = 0
  gitcommit_issue_icon = ''
  gitcommit_pr_icon = ''
  github_username = ''
  guifont = ''
  hiddenfileinfo = 1
  home_files_number = 6
  hosts_url = 'https://raw.githubusercontent.com/racaljk/hosts/master/hosts'
  info_symbol = 'ⓘ'
  keep_server_alive = 1
  language = ''
  leader_guide_default_group_name = ''
  leader_guide_flatten = 1
  leader_guide_hspace = 5
  leader_guide_max_size = 0
  leader_guide_position = 'botright'
  leader_guide_run_map_on_popup = 1
  leader_guide_sort_horizontal = 0
  leader_guide_submode_mappings = {'<C-C>': 'win_close'}
  leader_guide_vertical = 0
  lint_engine = 'neomake'
  lint_on_save = 1
  lint_on_the_fly = 0
  max_column = 120
  plugin_bundle_dir = '/home/shogun/.cache/vimfiles/'
  plugin_manager = 'dein'
  plugin_manager_processes = 16
  plugin_name = 'open-browser.vim'
  project_auto_root = 1
  project_non_root = ''
  project_rooter_outermost = 1
  project_rooter_patterns = ['.git/', '_darcs/', '.hg/', '.bzr/', '.svn/']
  projects_cache_num = 20
  realtime_leader_guide = 1
  relativenumber = 1
  search_tools = ['rg', 'ag', 'pt', 'ack', 'grep', 'findstr', 'git']
  sidebar_direction = ''
  sidebar_width = 30
  smartcloseignoreft = ['tagbar', 'vimfiler', 'defx', 'NvimTree', 'SpaceVimRunner', 'SpaceVimREPL', 'SpaceVimQuickFix', 'HelpDescribe', 'VebuggerShell', 'VebuggerTerminal', 'SpaceVimTabsManager']
  smartcloseignorewin = ['__Tagbar__', 'vimfiler:default']
  snippet_engine = 'neosnippet'
  src_root = 'E:\sources\'
  statusline_iseparator = 'nil'
  statusline_left = ['winnr', 'filename', 'major mode', 'search count', 'syntax checking', 'minor mode lighters']
  statusline_right = ['fileformat', 'cursorpos', 'percentage']
  statusline_separator = 'nil'
  statusline_unicode = 1
  terminal_cursor_shape = 2
  todo_labels = ['fixme', 'question', 'todo', 'idea']
  todo_prefix = '@'
  update_retry_cnt = 3
  version = '2.1.0-dev'
  vim_help_language = 'en'
  vimcompatible = 0
  warning_symbol = '⚠'
  wildignore = '*/tmp/*,*.so,*.swp,*.zip,*.class,tags,*.jpg,*.ttf,*.TTF,*.png,*/target/*,.git,.svn,.hg,.DS_Store,*.svg'
  windisk_encoding = 'cp936'
  windows_index_type = 3
  windows_leader = 's'
  windows_smartclose = 'q'
  wrap_line = 0
```

### SpaceVim layers :

```toml
[[layers]]
  name="autocomplete"
[[layers]]
  name="checkers"
[[layers]]
  name="format"
[[layers]]
  name="edit"
[[layers]]
  name="ui"
[[layers]]
  name="core"
[[layers]]
  name="core#banner"
[[layers]]
  name="core#statusline"
[[layers]]
  name="core#tabline"
```

### SpaceVim Health checking :

SpaceVim clipboard support check report:
Checking +clipboard:
      SUCCEED!
SpaceVim environment check report:
Current progpath: nvim(/opt/nvim/bin/nvim)
version: NVIM v0.8.0-1210-gd367ed9b2
OS: linux
[shell, shellcmdflag, shellslash]: ['/usr/bin/zsh', '-c', 0]
SpaceVim lua support check report:
Checking +lua:
      Failed : Known issue, neovim do not support lua now.
SpaceVim python support check report:
Checking +python3:
      SUCCEED!
Checking +python:
      Failed : to support +python, you need run `pip2 install neovim`
autocomplete layer health:

ok

checkers layer health:

ok

format layer health:

ok

edit layer health:

ok

ui layer health:

ok

core layer health:

ok

core#banner layer health:

ok

core#statusline layer health:

ok

core#tabline layer health:

ok

### SpaceVim runtime log :

```log
[ SpaceVim ] : logger file  does not exists, only log for current process will be shown!
[ SpaceVim ] [18:46:49] [0.000] [ Info  ] Loading SpaceVim from: /home/shogun/.SpaceVim/
[ SpaceVim ] [18:46:49] [0.065] [ Info  ] Startup with argv: ['nvim', '.zshrc']
[ SpaceVim ] [18:46:49] [0.065] [ Info  ] init default vim options
[ SpaceVim ] [18:46:49] [0.065] [ Info  ] options init done
[ SpaceVim ] [18:46:49] [0.069] [ Info  ] start loading global config >>>
[ SpaceVim ] [18:46:49] [0.069] [ Info  ] global_dir is: /home/shogun/.SpaceVim.d/
[ SpaceVim ] [18:46:49] [0.069] [ Info  ] find global config: /home/shogun/.SpaceVim.d/init.toml
[ SpaceVim ] [18:46:49] [0.072] [ Info  ] SpaceVim server startup at:/tmp/spacevim_nvim_server
[ SpaceVim ] [18:46:49] [0.092] [ Info  ] the func has been added to mode:syntax-checking
[ SpaceVim ] [18:46:49] [0.110] [ Info  ] the func has been added to mode:spell-checking
[ SpaceVim ] [18:46:49] [0.110] [ Info  ] the func has been added to mode:paste-mode
[ SpaceVim ] [18:46:49] [0.134] [ Info  ] finished loading SpaceVim!
[  project ] [18:46:49] [0.175] [ Info  ] Load projects cache from: /home/shogun/.cache/SpaceVim/projects.json
[ SpaceVim ] [18:48:24] [95.087] [ Info  ] the func has been added to mode:syntax-checking
[ SpaceVim ] [18:48:24] [95.092] [ Info  ] the func has been added to mode:spell-checking
[ SpaceVim ] [18:48:24] [95.092] [ Info  ] the func has been added to mode:paste-mode

```
</details>

## Screenshots
![image](https://user-images.githubusercontent.com/4404023/193411410-6f717c99-0921-44a0-84f7-dbc64bd20804.png)

Running `:SPDebugInfo!` : 
![image](https://user-images.githubusercontent.com/4404023/193411434-bc196882-5d31-4f6c-84c4-586f862bdc92.png)

**Comment by the-moisrex** [comment]
This gets resolved by this: #4718

**Comment by wsdjeg** [comment]
thanks
